### PR TITLE
This commit completes Task 11 GET /api/articles (topic query)

### PR DESCRIPTION
**Pull request by yaiqbal**

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -108,6 +108,38 @@ describe('/api/articles', () => {
...
+          expect(article).toMatchObject({
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
+            author: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url : expect.any(String),
+            body: expect.any(String)
+          })
```
Comment (MKD01) [inline __tests__/index.test.js:129]: We should be asserting that every topic is mitch here otherwise there is no way of actually knowing if the filter worked.

File: __tests__/index.test.js [hunk 2]
```diff
@@ -108,6 +108,38 @@ describe('/api/articles', () => {
...
+  test('GET:404 sends an appropriate status and error message when queried for non-existent topic (valid datatype)', () => {
+    const topic = "northcoders"
+    return request(app)
+      .get('/api/articles')
+      .query({topic})
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe('Topic not found');
+      });
+  });
```
Comment (MKD01) [inline __tests__/index.test.js:142]: Nice error handling test

File: __tests__/index.test.js [hunk 3]
```diff
@@ -108,6 +108,38 @@ describe('/api/articles', () => {
...
+            topic: expect.any(String),
+            author: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url : expect.any(String),
+            body: expect.any(String)
+          })
+        });
+      });
+  });
```
Comment (MKD01) [inline __tests__/index.test.js:132]: Another test that should be added here is a test to ensure when given a topic that does exist but has no associated articles we still receive an empty array.

**Review by MKD01 — COMMENTED** [review]
Really good start to this. There is a missing test and some minor changes I would like to see before you move on from this pr so please re-submit this pr once those changes have been added

File: __tests__/index.test.js (hunk 1 above)
Comment (yaiqbal) [reply __tests__/index.test.js:129]: Added this check. This change will be part of re-commit of  Task 11.

File: __tests__/index.test.js (hunk 3 above)
Comment (yaiqbal) [reply __tests__/index.test.js:132]: Added this test.  This change will be part of re-commit of Task 11.
